app/src/main/kotlin/com/wire/android/ui/common/SearchBar.kt

```diff
@@ -0,0 +1,165 @@
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.LocalContentAlpha
+import androidx.compose.material.LocalContentColor
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.OutlinedTextField
+import androidx.compose.material.Text
+import androidx.compose.material.TextFieldDefaults
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment.Companion.CenterHorizontally
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.graphics.Color.Companion.Transparent
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.wire.android.R
+import com.wire.android.ui.theme.WireColor
+
+@Suppress("LongParameterList")
+@ExperimentalAnimationApi
+@ExperimentalComposeUiApi
+@Composable
+fun SearchBarFullScreen(
+    searchText: String,
+    placeholderText: String = "",
+    onSearchTextChanged: (String) -> Unit = {},
+    onClearClick: () -> Unit = {},
+    onNavigateBack: () -> Unit = {},
+    matchesFound: Boolean,
+    modifier: Modifier = Modifier,
+    results: @Composable () -> Unit = {}
+) {
+    var showClearButton by remember { mutableStateOf(false) }
+    val keyboardController = LocalSoftwareKeyboardController.current
+    val focusRequester = remember { FocusRequester() }
+
+    TopAppBar(title = { Text("") },
+        backgroundColor = MaterialTheme.colors.background,
+        contentColor = MaterialTheme.colors.onBackground,
+        navigationIcon = {
+            IconButton(onClick = { onNavigateBack() }) {
+                Icon(
+                    imageVector = Icons.Filled.ArrowBack,
+                    contentDescription = stringResource(id = R.string.content_description_search_back)
+                )
+            }
+        }, actions = {
+            OutlinedTextField(
+                modifier = modifier
+                    .fillMaxWidth()
+                    .onFocusChanged { focusState ->
+                        showClearButton = (focusState.isFocused)
```

Comment:
i think this icon should become visible not `onFocusChanged`, but when user typed something into search. So it's like "remove everything"